app/server/Main.hs

```diff
@@ -56,12 +81,33 @@ main = do
     args <- parseArgsOrExit cli =<< getArgs
     when (args `isPresent` (longOption "help")) $ exitWithUsage cli
 
-    networkName <- getArg args cli (longOption "network") (decode @Network)
-    bridgePort <- getArg args cli (longOption "http-bridge-port") decode
-    _ <- getArg args cli (longOption "wallet-server-port") (decode @(Port "wallet"))
+    networkName <- getArg @String args cli (longOption "network") (decode @_ @Network)
+    bridgePort <- getArg @String args cli (longOption "http-bridge-port") decode
+    _ <- getArg @String args cli (longOption "wallet-server-port") (decode @_ @Int)
+    mnemonic <- getArg @[String] args cli (argument "mnemonic") decode
+
+    let wid = WalletId "MyWallet"
 
     network <- HttpBridge.newNetworkLayer (T.pack . encode $ networkName) bridgePort
-    listen network logBlock
+    db <- MVar.newDBLayer
+
+    putCheckpoints db (PrimaryKey wid) (mkWallet mnemonic :| [])
+    listen network $ \blocks -> do
```

Comment:
Can this live in the library and not the executable?

Specifically asking because I thought it would be nice to start the wallet / wallet components from ghci, and have the executable do little more than read CLI args and pass in.

Reply:
https://github.com/input-output-hk/cardano-wallet/pull/73#discussion_r266362881 applies

Reply:
That's right. Cf previous comment. This was handy to quickly test things out but I am not going to keep this as such.